### DISCUSSION TITLE
How to access static files from serverless functions?

### DISCUSSION THREAD

**Discussion by PierBover** [vercel/vercel#6739] [Help]
I have a function that returns some HTML and needs to add dynamic `<script>` tags because the JS and CSS assets are hashed (eg: `fe674a0d7dfdf7e582f2.js`).

What would be the best approach to solve this?

**Comment by delbaoliveira** [comment]
Hey @PierBover,

Could you please expand on your use case for using hashed file names? For example, would it be for cache invalidation? 

An alternative approach you may want to consider is to use the [Node `fs` module](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions) to access the files from a serverless function and the [`includeFiles` configuration option](https://vercel.com/docs/runtimes#advanced-usage/technical-details/including-additional-files) which accepts a glob for files with wildcard characters.

**Reply by PierBover** [reply]
Hi @delbaoliveira thanks for your answer.

Yes, hashed filenames are standard practice for cache invalidation. Is this unnecessary for static files on Vercel?

Wouldn't including the files slow the functions start time?

Isn't there some kind of manifest with all the project files that the functions can check?

**Reply by delbaoliveira** [reply]
Thank you! 

If all you’re trying to achieve is cache invalidation for static files, there is no manual step required with Vercel. Here's more information: https://vercel.com/docs/edge-network/caching#static-files

**Reply by PierBover** [reply]
Awesome!

Is there a way to rewrite headers just for static files?
